packages/ember-metal/lib/meta.js

```diff
@@ -183,27 +183,55 @@ export class Meta {
     return this[key] || (this[key] = Object.create(null));
   }
 
-  _getInherited(key) {
+  _forEachInherited() {
```

Comment:
accepts list of keys, when `obj.prop1.prop2.prop3` `_forEachInherited('obj','prop1','prop2','prop3', fn)` will call fn with value of `prop3` with current meta and its parent